plugins/Hosting/js/sagas/saga.js

```diff
@@ -0,0 +1,233 @@
+import { takeEvery } from 'redux-saga'
+import { put, take, fork } from 'redux-saga/effects'
+import * as actions from '../actions/actions.js'
+import * as constants from '../constants/constants.js'
+import * as helper from '../utils/host.js'
+import { Map, List } from 'immutable'
+import BigNumber from 'bignumber.js'
+
+// siadCall: promisify Siad API calls.  Resolve the promise with `response` if the call was successful,
+// otherwise reject the promise with `err`.
+const siadCall = (uri) => new Promise((resolve, reject) => {
+	SiaAPI.call(uri, (err, response) => {
+		if (err) {
+			reject({ message: err })
+		} else {
+			resolve(response)
+		}
+	})
+})
+
+const fetchStorageFiles = () => new Promise((resolve, reject) => {
+	siadCall({
+		url: '/storage',
+		method: 'GET',
+	}).then((fetchedFiles) => {
+		resolve(List(fetchedFiles.StorageFolderMetadata).map((file) => (
+			Map({
+				path: file.path,
+				size: (new BigNumber(file.capacity)).times('1e-9').toString(),
+				free: (new BigNumber(file.capacityremaining)).times('1e-9').toString(),
+			})
+		)).toList())
```

Comment:
don't bother converting to and from an `immutable.List` within the same expression. Immutable objects are only really helpful if you're passing them around.